server/src/controllers/cars.ts

```diff
@@ -130,11 +131,17 @@ const getFilteredCars = async (req: Request) => {
 
 //-------------------------------------------------------
 
+// eslint-disable-next-line consistent-return
 const updateCars = async (req: Request) => {
   const { body } = req;
   const { id } = req.params;
   await updateCarSchema.validate(body);
-  const result = await updateCarServes(body, id);
+  const getCar = await getCarInfo(id);
+  if (getCar[0].state === 'under-check' || getCar[0].state === 'on-market') {
```

Comment:
Not sure why you check car `state`, If you want only to update cars with `under-check`
check if not state throw error and then do whatever controller should do, this will be more readable